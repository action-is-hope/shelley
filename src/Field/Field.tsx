import type React from "react";
import {
  ReactNode,
  ReactElement,
  cloneElement,
  isValidElement,
  Children,
  forwardRef,
  HTMLProps,
} from "react";
import type {
  Volume,
  FieldVariants,
  ComponentBase,
  LabelPosition,
} from "../typings/shared-types";
import type { Validation } from "../typings/shared-types";
import { Label } from "../Label";
import { HelpText } from "../HelpText";
import { VisuallyHidden } from "../VisuallyHidden";
import { InputAdornment } from "../InputAdornment";
import { st, classes } from "./field.st.css";

export interface FieldContainerProps
  extends HTMLProps<HTMLDivElement>,
    ComponentBase {}
export interface FieldProps extends Validation, ComponentBase {
  /** Provide an error message that triggers the stylable error state. */
  errorMessage?: ReactNode;
  /** Provide some description or hint text to the field. */
  description?: ReactNode;
  /** Place a component so as to appear inside the TextInput start. */
  startAdornment?: ReactNode;
  /** Place a component so as to appear inside the TextInput end. */
  endAdornment?: ReactNode;
  /** The label to associated with the input. */
  label: ReactNode;
  /**
   * Position of the label.
   * @default "over"
   */
  labelPosition?: LabelPosition;
  /**
   * Disable the label transition.
   * @default false
   */
  disableLabelTransition?: boolean;
  /** Props for the label element. */
  labelProps?: Omit<React.HTMLProps<HTMLLabelElement>, "ref">;
  /**
   * Variant index.
   * @default "outlined"
   */
  variant?: FieldVariants;
  /**
   * Defines how 'loud' the field should be.
   * @default 1
   */
  vol?: Volume;
  /** Does the containing input have a value. */
  hasValue?: boolean;
  /** Props for the help text description element. */
  descriptionProps?: HTMLProps<HTMLDivElement>;
  /** Props for the help text error message element. */
  errorMessageProps?: HTMLProps<HTMLDivElement>;
  /** Props for the field container. */
  fieldContainerProps?: FieldContainerProps;
  /** Enable disabled state. */
  isDisabled?: boolean;
  isReadOnly?: boolean;
}

interface FieldInternalProps
  extends Pick<
      React.HTMLProps<HTMLDivElement>,
      Exclude<keyof React.HTMLProps<HTMLDivElement>, "label">
    >,
    FieldProps {}

function Field(props: FieldInternalProps, ref?: React.Ref<HTMLDivElement>) {
  const {
    className: classNameProp,
    children,
    errorMessage,
    validationState,
    startAdornment,
    endAdornment,
    description,
    label: labelStringProp,
    labelProps,
    labelPosition = "over",
    descriptionProps,
    errorMessageProps,
    disableLabelTransition = false,
    variant = "outlined",
    hasValue: hasValueProp,
    fieldContainerProps,
    isReadOnly,
    isRequired,
    isDisabled,
    vol = 1,
    "data-id": dataId,
    ...rest
  } = props;

  const hasValue =
    disableLabelTransition || startAdornment ? true : hasValueProp;

  const error = validationState === "invalid";
  const childrenWithProps = Children.map(children, (child) => {
    if (isValidElement(child)) {
      return cloneElement(child as ReactElement, {
        className: st(
          classes.fieldInput,
          (child?.props as React.HTMLProps<HTMLElement>)?.className
        ),
      });
    } else return;
  });

  const label = labelStringProp && (
    <Label
      className={classes.inputLabel}
      data-id={dataId ? `${dataId}--label` : undefined}
      {...labelProps}
    >
      {labelStringProp}
    </Label>
  );
  const hideLabel = labelPosition === "hidden";
  return (
    <div
      className={st(
        classes.root,
        {
          hasValue,
          error,
          isDisabled,
          isRequired,
          isReadOnly,
          variant: variant || undefined,
          vol: vol !== false ? vol : undefined,
          labelPosition,
        },
        classNameProp
      )}
      data-id={dataId}
      ref={ref}
      {...rest}
    >
      {hideLabel && label ? <VisuallyHidden>{label}</VisuallyHidden> : label}

      <div
        {...fieldContainerProps}
        className={st(classes.fieldContainer, fieldContainerProps?.className)}
      >
        {typeof startAdornment === "string" ? (
          <InputAdornment
            className={classes.startAdornment}
            data-id={dataId ? `${dataId}--start-adornment` : undefined}
          >
            {startAdornment}
          </InputAdornment>
        ) : (
          startAdornment
        )}
        {childrenWithProps}
        {typeof endAdornment === "string"
          ? endAdornment && (
              <InputAdornment
                className={classes.endAdornment}
                data-id={dataId ? `${dataId}--end-adornment` : undefined}
              >
                {endAdornment}
              </InputAdornment>
            )
          : endAdornment}
        {/*  */}
        {variant && (
          <fieldset aria-hidden="true" className={classes.fieldset}>
            <legend className={classes.legend}>
              {!hideLabel && labelStringProp}
            </legend>
          </fieldset>
        )}
      </div>

      <HelpText
        className={classes.helpText}
        {...{
          description,
          descriptionProps,
          errorMessage,
          errorMessageProps,
          validationState,
        }}
        data-id={dataId ? `${dataId}--helpText` : undefined}
      />
    </div>
  );
}

const _Field = forwardRef(Field);
export { _Field as Field };
