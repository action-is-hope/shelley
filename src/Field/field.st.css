@st-namespace "field";
@st-import HelpText from "../HelpText/helpText.st.css";
@st-import InputLabel from "../Label/label.st.css";

.root {
  -st-states:
    isRequired,
    isDisabled,
    isReadOnly,
    hasValue,
    error,
    variant(string),
    vol(number),
    disableLabelTransition,
    labelPosition(enum(top, side, over, hidden));
}

.root {}

.inputLabel {
  -st-extends: InputLabel;
}

.fieldContainer {}

.startAdornment {}

.endAdornment {}

.textAreaWrap {}

.fieldInput {}

.helpText {
  -st-extends: HelpText;
}

.fieldset {}

.legend {}
