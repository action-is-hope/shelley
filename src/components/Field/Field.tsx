import type React from "react";
import {
  ReactNode,
  ReactElement,
  cloneElement,
  isValidElement,
  Children,
  forwardRef,
  HTMLProps,
} from "react";
import type {
  Volume,
  FieldVariants,
  ComponentBase,
  LabelPosition,
} from "../types";
import type { Validation } from "../../typings/shared-types";
import Label from "../Label/Label";
import { HelpText } from "../HelpText/HelpText";
import VisuallyHidden from "../VisuallyHidden/VisuallyHidden";
import InputAdornment from "../InputAdornment/InputAdornment";
/* = Style API. */
import { st, classes } from "./field.st.css";

export interface FieldProps extends Validation, ComponentBase {
  /** Provide an error message that triggers the stylable error state. */
  errorMessage?: ReactNode;
  /** Provide some description or hint text to the field. */
  description?: ReactNode;
  /** Place a component so as to appear inside the TextInput start. */
  startAdornment?: ReactNode;
  /** Place a component so as to appear inside the TextInput end. */
  endAdornment?: ReactNode;
  /** The label to associated with the input. */
  label: ReactNode;
  /**
   * Position of the label.
   * @default "over"
   */
  labelPosition?: LabelPosition;
  /**
   * Disable the label transition.
   * @default false
   */
  disableLabelTransition?: boolean;
  /** Props for the label element. */
  labelProps?: React.HTMLProps<HTMLLabelElement>;
  /**
   * Variant index.
   * @default "outlined"
   */
  variant?: FieldVariants;
  /**
   * Defines how 'loud' the field should be.
   * @default 1
   */
  vol?: Volume;
  /** Does the containing input have a value. */
  hasValue?: boolean;
  /** Props for the help text description element. */
  descriptionProps?: HTMLProps<HTMLDivElement>;
  /** Props for the help text error message element. */
  errorMessageProps?: HTMLProps<HTMLDivElement>;
  /** Props for the field container. */
  fieldContainerProps?: HTMLProps<HTMLDivElement>;
  /** Enable disabled state. */
  isDisabled?: boolean;
  isReadOnly?: boolean;
}

interface FieldInternalProps
  extends Pick<
      React.HTMLProps<HTMLDivElement>,
      Exclude<keyof React.HTMLProps<HTMLDivElement>, "label">
    >,
    FieldProps {}

function Field(props: FieldInternalProps, ref?: React.Ref<HTMLDivElement>) {
  const {
    className: classNameProp,
    children,
    errorMessage,
    validationState,
    startAdornment,
    endAdornment,
    description,
    label: labelStringProp,
    labelProps,
    labelPosition = "over",
    descriptionProps,
    errorMessageProps,
    disableLabelTransition = false,
    variant = "outlined",
    hasValue: hasValueProp,
    fieldContainerProps,
    isReadOnly,
    isRequired,
    isDisabled,
    vol = 1,
    "data-id": dataId,
    ...rest
  } = props;

  const hasValue =
    disableLabelTransition || startAdornment ? true : hasValueProp;

  const error = validationState === "invalid";
  const childrenWithProps = Children.map(children, (child) => {
    if (isValidElement(child)) {
      return cloneElement(child as ReactElement, {
        className: st(
          classes.fieldInput,
          (child?.props as React.HTMLProps<HTMLElement>)?.className
        ),
      });
    } else return;
  });
  // @todo empty labelStringProp should not render label
  const label = (
    <Label
      className={classes.inputLabel}
      data-id={dataId ? `${dataId}--label` : undefined}
      {...labelProps}
    >
      {labelStringProp}
    </Label>
  );
  const hideLabel = labelPosition === "hidden";
  return (
    <div
      className={st(
        classes.root,
        {
          hasValue,
          error,
          isDisabled,
          isRequired,
          isReadOnly,
          variant: variant || undefined,
          vol: vol !== false ? vol : undefined,
          labelPosition,
        },
        classNameProp
      )}
      data-id={dataId}
      ref={ref}
      {...rest}
    >
      {hideLabel ? <VisuallyHidden>{label}</VisuallyHidden> : label}

      <div
        {...fieldContainerProps}
        className={st(classes.fieldContainer, fieldContainerProps?.className)}
      >
        {startAdornment && (
          <InputAdornment
            className={classes.startAdornment}
            data-id={dataId ? `${dataId}--start-adornment` : undefined}
          >
            {startAdornment}
          </InputAdornment>
        )}
        {childrenWithProps}
        {endAdornment && (
          <InputAdornment
            className={classes.endAdornment}
            data-id={dataId ? `${dataId}--end-adornment` : undefined}
          >
            {endAdornment}
          </InputAdornment>
        )}
        {/*  */}
        {variant && (
          <fieldset aria-hidden="true" className={classes.fieldset}>
            <legend className={classes.legend}>
              {!hideLabel && labelStringProp}
            </legend>
          </fieldset>
        )}
      </div>

      <HelpText
        className={classes.helpText}
        {...{
          description,
          descriptionProps,
          errorMessage,
          errorMessageProps,
          validationState,
        }}
        data-id={dataId ? `${dataId}--helpText` : undefined}
      />
    </div>
  );
}

const _Field = forwardRef(Field);
export { _Field as Field };
