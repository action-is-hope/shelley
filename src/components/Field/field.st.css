/** field.st.css */
@st-import HelpText from "../HelpText/helpText.st.css";
@st-import InputLabel from "../Label/label.st.css";

@namespace "Field";

.root {
  -st-states:
    isRequired,
    isDisabled,
    isReadOnly,
    hasValue,
    error,
    variant(string),
    vol(number),
    disableLabelTransition,
    labelPosition(enum(top, side, over, hidden));
}

.root {
  display: grid;
  position: relative;
}

.inputLabel {
  -st-extends: InputLabel;
}

.fieldContainer {
  position: relative;
  display: inline-flex;
}

.startAdornment {}

.endAdornment {}

.textAreaWrap {}

.fieldInput {}

.helpText {
  -st-extends: HelpText;
}

/* Basic grid setup */
.root:labelPosition(top) {
  grid-template-columns: 1fr;
}

.root:labelPosition(side) {
  grid-template-columns: .35fr 1fr;
}

.root:labelPosition(side)::helpText {
  grid-area: 2 / 2;
}

.root:labelPosition(over) {
  grid-template-columns: 1fr;
}

.root:labelPosition(hidden) {
  grid-template-columns: 1fr;
}

/* Transition label... */
.fieldset {
  text-align: left;
  position: absolute;
  inset: -5px 0px 0px;
  margin: 0px;
  padding: 0px 8px;
  pointer-events: none;
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  overflow: hidden;
  min-width: 0%;
}

.legend {
  float: unset;
  width: auto;
  overflow: hidden;
  display: block;
  padding: 0;
  height: 11px;
  font-size: 0.75em;
  visibility: hidden;
  max-width: 0.01px;
  white-space: nowrap;
  font-weight: 400;
}

.root:labelPosition(over):focus-within::legend,
.root:labelPosition(over):hasValue::legend {
  max-width: 100%;
  padding: 0 7px;
}

.root:labelPosition(over)::inputLabel {
  display: block;
  transform-origin: left top;
  position: absolute;
  left: 0;
  top: 0;

  /* @todo variable... for transition */
  transition: color 200ms cubic-bezier(0, 0, 0.2, 1) 0ms, transform 200ms cubic-bezier(0, 0, 0.2, 1) 0ms, max-width 200ms cubic-bezier(0, 0, 0.2, 1) 0ms;
  z-index: 1;

  cursor: text;
  max-width: calc(100% - 3em);
}

.root:labelPosition(over)::inputLabel::labelText {
  display: block;
  transform-origin: 0;
  transition: transform 200ms cubic-bezier(0, 0, 0.2, 1) 0ms;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.root:labelPosition(over):focus-within::inputLabel,
.root:labelPosition(over):hasValue::inputLabel {
  transform: translate(0, -10px);
}

.root:labelPosition(over):focus-within::inputLabel::labelText,
.root:labelPosition(over):hasValue::inputLabel::labelText {
  transform: scale(0.75);
}

.root:labelPosition(over)::fieldInput::placeholder,
.root:labelPosition(over)::fieldInput textarea::placeholder {
  transition: opacity 200ms cubic-bezier(0, 0, 0.2, 1) 0ms;
  opacity: 0;
}

.root:labelPosition(over)::fieldInput:focus::placeholder,
.root:labelPosition(over):hasValue::fieldInput::placeholder,
.root:labelPosition(over)::fieldInput textarea:focus::placeholder,
.root:labelPosition(over):hasValue::fieldInput textarea::placeholder {
  opacity: 1;
}
