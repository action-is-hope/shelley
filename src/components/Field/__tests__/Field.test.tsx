import renderer from "react-test-renderer";
import Field from "../Field";

const fieldPropsTest = {
  "data-id": "field",
  label: "My field",
  labelProps: {
    id: "label-id",
    htmlFor: "test",
    data: "random",
  },
};

const sampleField = (
  <input id="testField" type="text" aria-labelledby="label-id" />
);

describe("Field", () => {
  it("Renders default field as variant 'outlined', labelPosition 'over', volume '1'", () => {
    const tree = renderer
      .create(<Field {...fieldPropsTest}>{sampleField}</Field>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renderd as variant 'outlined'", () => {
    const tree = renderer
      .create(
        <Field {...fieldPropsTest} variant="outlined">
          {sampleField}
        </Field>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renderd as variant 'filled'", () => {
    const tree = renderer
      .create(
        <Field {...fieldPropsTest} variant="filled">
          {sampleField}
        </Field>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders as variant 'quiet'", () => {
    const tree = renderer
      .create(
        <Field {...fieldPropsTest} variant="quiet">
          {sampleField}
        </Field>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders with no variant 'false'", () => {
    const tree = renderer
      .create(
        <Field {...fieldPropsTest} variant={false}>
          {sampleField}
        </Field>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders with labelPosition 'top'", () => {
    const tree = renderer
      .create(
        <Field {...fieldPropsTest} labelPosition="top">
          {sampleField}
        </Field>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders with labelPosition 'side'", () => {
    const tree = renderer
      .create(
        <Field {...fieldPropsTest} labelPosition="side">
          {sampleField}
        </Field>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders with labelPosition 'over'", () => {
    const tree = renderer
      .create(
        <Field {...fieldPropsTest} labelPosition="over">
          {sampleField}
        </Field>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders with labelPosition 'hidden'", () => {
    const tree = renderer
      .create(
        <Field {...fieldPropsTest} labelPosition="hidden">
          {sampleField}
        </Field>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders with volume '1'", () => {
    const tree = renderer
      .create(
        <Field {...fieldPropsTest} vol={1}>
          {sampleField}
        </Field>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders with volume '2'", () => {
    const tree = renderer
      .create(
        <Field {...fieldPropsTest} vol={2}>
          {sampleField}
        </Field>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders with volume '3'", () => {
    const tree = renderer
      .create(
        <Field {...fieldPropsTest} vol={3}>
          {sampleField}
        </Field>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders with volume '4'", () => {
    const tree = renderer
      .create(
        <Field {...fieldPropsTest} vol={4}>
          {sampleField}
        </Field>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders with volume '5'", () => {
    const tree = renderer
      .create(
        <Field {...fieldPropsTest} vol={5}>
          {sampleField}
        </Field>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders with volume '6'", () => {
    const tree = renderer
      .create(
        <Field {...fieldPropsTest} vol={6}>
          {sampleField}
        </Field>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders with no volume 'undefined'", () => {
    const tree = renderer
      .create(
        <Field {...fieldPropsTest} vol={undefined}>
          {sampleField}
        </Field>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
