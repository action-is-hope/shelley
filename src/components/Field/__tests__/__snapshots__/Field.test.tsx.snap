// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Renderd as variant 'filled' 1`] = `
<div
  className="Field2808104046__root Field2808104046---variant-6-filled Field2808104046---vol-1-1 Field2808104046---labelPosition-4-over"
  data-id="field"
>
  <label
    className="Label3203939847__root Field2808104046__inputLabel"
    data="random"
    htmlFor="test"
    id="label-id"
  >
    <span
      className="Label3203939847__labelText"
    >
      My field
    </span>
  </label>
  <div
    className="Field2808104046__fieldContainer"
  >
    <input
      aria-labelledby="label-id"
      className="Field2808104046__fieldInput"
      id="testField"
      type="text"
    />
    <fieldset
      aria-hidden="true"
      className="Field2808104046__fieldset"
    >
      <legend
        className="Field2808104046__legend"
      >
        My field
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Field Renderd as variant 'outlined' 1`] = `
<div
  className="Field2808104046__root Field2808104046---variant-8-outlined Field2808104046---vol-1-1 Field2808104046---labelPosition-4-over"
  data-id="field"
>
  <label
    className="Label3203939847__root Field2808104046__inputLabel"
    data="random"
    htmlFor="test"
    id="label-id"
  >
    <span
      className="Label3203939847__labelText"
    >
      My field
    </span>
  </label>
  <div
    className="Field2808104046__fieldContainer"
  >
    <input
      aria-labelledby="label-id"
      className="Field2808104046__fieldInput"
      id="testField"
      type="text"
    />
    <fieldset
      aria-hidden="true"
      className="Field2808104046__fieldset"
    >
      <legend
        className="Field2808104046__legend"
      >
        My field
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Field Renders as variant 'quiet' 1`] = `
<div
  className="Field2808104046__root Field2808104046---variant-5-quiet Field2808104046---vol-1-1 Field2808104046---labelPosition-4-over"
  data-id="field"
>
  <label
    className="Label3203939847__root Field2808104046__inputLabel"
    data="random"
    htmlFor="test"
    id="label-id"
  >
    <span
      className="Label3203939847__labelText"
    >
      My field
    </span>
  </label>
  <div
    className="Field2808104046__fieldContainer"
  >
    <input
      aria-labelledby="label-id"
      className="Field2808104046__fieldInput"
      id="testField"
      type="text"
    />
    <fieldset
      aria-hidden="true"
      className="Field2808104046__fieldset"
    >
      <legend
        className="Field2808104046__legend"
      >
        My field
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Field Renders default field as variant 'outlined', labelPosition 'over', volume '1' 1`] = `
<div
  className="Field2808104046__root Field2808104046---variant-8-outlined Field2808104046---vol-1-1 Field2808104046---labelPosition-4-over"
  data-id="field"
>
  <label
    className="Label3203939847__root Field2808104046__inputLabel"
    data="random"
    htmlFor="test"
    id="label-id"
  >
    <span
      className="Label3203939847__labelText"
    >
      My field
    </span>
  </label>
  <div
    className="Field2808104046__fieldContainer"
  >
    <input
      aria-labelledby="label-id"
      className="Field2808104046__fieldInput"
      id="testField"
      type="text"
    />
    <fieldset
      aria-hidden="true"
      className="Field2808104046__fieldset"
    >
      <legend
        className="Field2808104046__legend"
      >
        My field
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Field Renders with labelPosition 'hidden' 1`] = `
<div
  className="Field2808104046__root Field2808104046---variant-8-outlined Field2808104046---vol-1-1 Field2808104046---labelPosition-6-hidden"
  data-id="field"
>
  <span
    className="visuallyHidden519588831__root"
  >
    <label
      className="Label3203939847__root Field2808104046__inputLabel"
      data="random"
      htmlFor="test"
      id="label-id"
    >
      <span
        className="Label3203939847__labelText"
      >
        My field
      </span>
    </label>
  </span>
  <div
    className="Field2808104046__fieldContainer"
  >
    <input
      aria-labelledby="label-id"
      className="Field2808104046__fieldInput"
      id="testField"
      type="text"
    />
    <fieldset
      aria-hidden="true"
      className="Field2808104046__fieldset"
    >
      <legend
        className="Field2808104046__legend"
      />
    </fieldset>
  </div>
</div>
`;

exports[`Field Renders with labelPosition 'over' 1`] = `
<div
  className="Field2808104046__root Field2808104046---variant-8-outlined Field2808104046---vol-1-1 Field2808104046---labelPosition-4-over"
  data-id="field"
>
  <label
    className="Label3203939847__root Field2808104046__inputLabel"
    data="random"
    htmlFor="test"
    id="label-id"
  >
    <span
      className="Label3203939847__labelText"
    >
      My field
    </span>
  </label>
  <div
    className="Field2808104046__fieldContainer"
  >
    <input
      aria-labelledby="label-id"
      className="Field2808104046__fieldInput"
      id="testField"
      type="text"
    />
    <fieldset
      aria-hidden="true"
      className="Field2808104046__fieldset"
    >
      <legend
        className="Field2808104046__legend"
      >
        My field
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Field Renders with labelPosition 'side' 1`] = `
<div
  className="Field2808104046__root Field2808104046---variant-8-outlined Field2808104046---vol-1-1 Field2808104046---labelPosition-4-side"
  data-id="field"
>
  <label
    className="Label3203939847__root Field2808104046__inputLabel"
    data="random"
    htmlFor="test"
    id="label-id"
  >
    <span
      className="Label3203939847__labelText"
    >
      My field
    </span>
  </label>
  <div
    className="Field2808104046__fieldContainer"
  >
    <input
      aria-labelledby="label-id"
      className="Field2808104046__fieldInput"
      id="testField"
      type="text"
    />
    <fieldset
      aria-hidden="true"
      className="Field2808104046__fieldset"
    >
      <legend
        className="Field2808104046__legend"
      >
        My field
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Field Renders with labelPosition 'top' 1`] = `
<div
  className="Field2808104046__root Field2808104046---variant-8-outlined Field2808104046---vol-1-1 Field2808104046---labelPosition-3-top"
  data-id="field"
>
  <label
    className="Label3203939847__root Field2808104046__inputLabel"
    data="random"
    htmlFor="test"
    id="label-id"
  >
    <span
      className="Label3203939847__labelText"
    >
      My field
    </span>
  </label>
  <div
    className="Field2808104046__fieldContainer"
  >
    <input
      aria-labelledby="label-id"
      className="Field2808104046__fieldInput"
      id="testField"
      type="text"
    />
    <fieldset
      aria-hidden="true"
      className="Field2808104046__fieldset"
    >
      <legend
        className="Field2808104046__legend"
      >
        My field
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Field Renders with no variant 'false' 1`] = `
<div
  className="Field2808104046__root Field2808104046---vol-1-1 Field2808104046---labelPosition-4-over"
  data-id="field"
>
  <label
    className="Label3203939847__root Field2808104046__inputLabel"
    data="random"
    htmlFor="test"
    id="label-id"
  >
    <span
      className="Label3203939847__labelText"
    >
      My field
    </span>
  </label>
  <div
    className="Field2808104046__fieldContainer"
  >
    <input
      aria-labelledby="label-id"
      className="Field2808104046__fieldInput"
      id="testField"
      type="text"
    />
  </div>
</div>
`;

exports[`Field Renders with no volume 'undefined' 1`] = `
<div
  className="Field2808104046__root Field2808104046---variant-8-outlined Field2808104046---vol-1-1 Field2808104046---labelPosition-4-over"
  data-id="field"
>
  <label
    className="Label3203939847__root Field2808104046__inputLabel"
    data="random"
    htmlFor="test"
    id="label-id"
  >
    <span
      className="Label3203939847__labelText"
    >
      My field
    </span>
  </label>
  <div
    className="Field2808104046__fieldContainer"
  >
    <input
      aria-labelledby="label-id"
      className="Field2808104046__fieldInput"
      id="testField"
      type="text"
    />
    <fieldset
      aria-hidden="true"
      className="Field2808104046__fieldset"
    >
      <legend
        className="Field2808104046__legend"
      >
        My field
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Field Renders with volume '1' 1`] = `
<div
  className="Field2808104046__root Field2808104046---variant-8-outlined Field2808104046---vol-1-1 Field2808104046---labelPosition-4-over"
  data-id="field"
>
  <label
    className="Label3203939847__root Field2808104046__inputLabel"
    data="random"
    htmlFor="test"
    id="label-id"
  >
    <span
      className="Label3203939847__labelText"
    >
      My field
    </span>
  </label>
  <div
    className="Field2808104046__fieldContainer"
  >
    <input
      aria-labelledby="label-id"
      className="Field2808104046__fieldInput"
      id="testField"
      type="text"
    />
    <fieldset
      aria-hidden="true"
      className="Field2808104046__fieldset"
    >
      <legend
        className="Field2808104046__legend"
      >
        My field
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Field Renders with volume '2' 1`] = `
<div
  className="Field2808104046__root Field2808104046---variant-8-outlined Field2808104046---vol-1-2 Field2808104046---labelPosition-4-over"
  data-id="field"
>
  <label
    className="Label3203939847__root Field2808104046__inputLabel"
    data="random"
    htmlFor="test"
    id="label-id"
  >
    <span
      className="Label3203939847__labelText"
    >
      My field
    </span>
  </label>
  <div
    className="Field2808104046__fieldContainer"
  >
    <input
      aria-labelledby="label-id"
      className="Field2808104046__fieldInput"
      id="testField"
      type="text"
    />
    <fieldset
      aria-hidden="true"
      className="Field2808104046__fieldset"
    >
      <legend
        className="Field2808104046__legend"
      >
        My field
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Field Renders with volume '3' 1`] = `
<div
  className="Field2808104046__root Field2808104046---variant-8-outlined Field2808104046---vol-1-3 Field2808104046---labelPosition-4-over"
  data-id="field"
>
  <label
    className="Label3203939847__root Field2808104046__inputLabel"
    data="random"
    htmlFor="test"
    id="label-id"
  >
    <span
      className="Label3203939847__labelText"
    >
      My field
    </span>
  </label>
  <div
    className="Field2808104046__fieldContainer"
  >
    <input
      aria-labelledby="label-id"
      className="Field2808104046__fieldInput"
      id="testField"
      type="text"
    />
    <fieldset
      aria-hidden="true"
      className="Field2808104046__fieldset"
    >
      <legend
        className="Field2808104046__legend"
      >
        My field
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Field Renders with volume '4' 1`] = `
<div
  className="Field2808104046__root Field2808104046---variant-8-outlined Field2808104046---vol-1-4 Field2808104046---labelPosition-4-over"
  data-id="field"
>
  <label
    className="Label3203939847__root Field2808104046__inputLabel"
    data="random"
    htmlFor="test"
    id="label-id"
  >
    <span
      className="Label3203939847__labelText"
    >
      My field
    </span>
  </label>
  <div
    className="Field2808104046__fieldContainer"
  >
    <input
      aria-labelledby="label-id"
      className="Field2808104046__fieldInput"
      id="testField"
      type="text"
    />
    <fieldset
      aria-hidden="true"
      className="Field2808104046__fieldset"
    >
      <legend
        className="Field2808104046__legend"
      >
        My field
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Field Renders with volume '5' 1`] = `
<div
  className="Field2808104046__root Field2808104046---variant-8-outlined Field2808104046---vol-1-5 Field2808104046---labelPosition-4-over"
  data-id="field"
>
  <label
    className="Label3203939847__root Field2808104046__inputLabel"
    data="random"
    htmlFor="test"
    id="label-id"
  >
    <span
      className="Label3203939847__labelText"
    >
      My field
    </span>
  </label>
  <div
    className="Field2808104046__fieldContainer"
  >
    <input
      aria-labelledby="label-id"
      className="Field2808104046__fieldInput"
      id="testField"
      type="text"
    />
    <fieldset
      aria-hidden="true"
      className="Field2808104046__fieldset"
    >
      <legend
        className="Field2808104046__legend"
      >
        My field
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Field Renders with volume '6' 1`] = `
<div
  className="Field2808104046__root Field2808104046---variant-8-outlined Field2808104046---vol-1-6 Field2808104046---labelPosition-4-over"
  data-id="field"
>
  <label
    className="Label3203939847__root Field2808104046__inputLabel"
    data="random"
    htmlFor="test"
    id="label-id"
  >
    <span
      className="Label3203939847__labelText"
    >
      My field
    </span>
  </label>
  <div
    className="Field2808104046__fieldContainer"
  >
    <input
      aria-labelledby="label-id"
      className="Field2808104046__fieldInput"
      id="testField"
      type="text"
    />
    <fieldset
      aria-hidden="true"
      className="Field2808104046__fieldset"
    >
      <legend
        className="Field2808104046__legend"
      >
        My field
      </legend>
    </fieldset>
  </div>
</div>
`;
